datashuttle/utils/utils.py

```diff
@@ -121,19 +123,68 @@ def log_and_raise_error_not_exists_or_not_yaml(path_to_config: Path) -> None:
         log_and_raise_error("The config file must be a YAML file.")
 
 
-def get_first_sub_ses_keys(all_names: List[str]) -> List[str]:
-    """
-    Assumes sub / ses name is in standard form with sub/ses label
-    in the second position e.g. sub-001_id-...
+# TODO: test this method
+
+
+@overload
+def get_values_from_bids_formatted_name(
+    all_names: List[str],
+    key: str,
+    return_as_int: Literal[True],
+    sort: bool = False,
+) -> List[int]:
+    ...
+
+
+@overload
+def get_values_from_bids_formatted_name(
+    all_names: List[str],
+    key: str,
+    return_as_int: Literal[False] = False,
+    sort: bool = False,
+) -> List[str]:
+    ...
 
-    Only look for folders / file with sub/ses in first key
-    to ignore all other files
+
+def get_values_from_bids_formatted_name(
+    all_names: List[str],
+    key: str,
+    return_as_int: bool = False,
+    sort: bool = False,
+) -> Union[List[int], List[str]]:
+    """
+    Find the values associated with a key from a list of all
+    BIDS-formatted file / folder names.
     """
-    return [
-        re.split("-|_", name)[1]
-        for name in all_names
-        if re.split("-|_", name)[0] in ["sub", "ses"]
-    ]
+    all_values = []
+    for name in all_names:
+
+        if key not in name:
+            raise_error(f"They key {key} is not found in {name}")
+
+        value = re.findall(f"{key}-(.*?)(?=_|$)", name)
```

Comment:
Could this regular expression be put into it's own function to enable it to be tested individually. Just thinking we may not notice if it changed slightly. Hopefully other tests would pick it up, but better safe than sorry.